pkgs/by-name/ht/httplz/package.nix

```diff
@@ -11,16 +11,17 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "httplz";
-  version = "1.13.2";
+  version = "2.3.0";
 
-  src = fetchCrate {
-    inherit version;
-    pname = "https";
-    hash = "sha256-uxEMgSrcxMZD/3GQuH9S/oYtMUPzgMR61ZzLcb65zXU=";
+  src = fetchFromGitHub {
+    owner = "thecoshman";
+    repo = "http";
+    rev = "v${version}";
+    hash = "sha256-qinhdpm9eaTdpUk4ZZLaH1D/CZ22k4RisHu8clZCEGo=";
   };
 
-  useFetchCargoVendor = true;
-  cargoHash = "sha256-DXSHaiiIRdyrlX4UYPFD3aTAv65k3x/PU2VW047odH0=";
+  cargoPatches = [ ./cargo-lock.patch ];
```

Comment:
https://nixos.org/manual/nixpkgs/unstable/#importing-a-cargo.lock-file

Reply:
Could you elaborate?

If I try setting `cargoLock.lockFile = ./Cargo.lock;`, I get an error and suggestion:

```
> Validating consistency between /build/source/Cargo.lock and /build/cargo-vendor-dir/Cargo.lock
> /nix/store/ibx4jfwlhjg4g0s6rrxrpaxa3ka8ns4m-diffutils-3.12/bin/diff: /build/source/Cargo.lock: No such file or directory
> ERROR: Missing Cargo.lock from src. Expected to find it at: /build/source/Cargo.lock
> Hint: You can use the cargoPatches attribute to add a Cargo.lock manually to the build.
```

Reply:
Try this?

> Setting cargoLock.lockFile or cargoLock.lockFileContents will not automatically add a Cargo.lock file to src. A straightforward solution is to use:

```nix
{
  postPatch = ''
    ln -s ${./Cargo.lock} Cargo.lock
  '';
}
```